source/fundamentals/transactions.txt

```diff
@@ -0,0 +1,162 @@
+.. _golang-transactions:
+
+============
+Transactions
+============
+
+.. default-domain:: mongodb
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+Overview
+--------
+
+In this guide, you can learn how to use **transactions** with the
+{+driver-long+}. A transaction is a unit of work that is composed of a series
+of operations. You can execute operations in a transaction so that they
+either succeed together or fail together when one or more of the
+operations fail.
+
+In MongoDB, transactions run within logical **sessions**. A session is a
+grouping of related read or write operations that you want to run
+sequentially. Sessions enable :manual:`causal consistency </core/read-isolation-consistency-recency/#causal-consistency>`
+for a group of operations or allow you to execute operations in an :website:`ACID
+transaction </basics/acid-transactions>`. MongoDB guarantees that the data
+involved in your transaction operations remains consistent, even if the operations
+encounter unexpected errors.
+
+With the {+driver-short+}, you can create a new session from a
+``Client`` instance as a ``Session`` type. You should only use a
+``Session`` instance created from a client to execute operations
+using that client instance or in a database or collection selected from that
+client.
+
+.. warning::
+   
+   You should not use custom implementations of the ``Session``
+   interface in production. Additionally, implementations of ``Session``
+   are not safe for concurrent use by multiple `goroutines
+   <https://www.golang-book.com/books/intro/10>`__.
```

Comment:
Optional: Of these two warnings, trying to use a `Session` across multiple goroutines is a much more common error that users make. Consider switching the order of these to warn about concurrent use of `Session` first.

```suggestion
   Implementations of ``Session`` are not safe for concurrent use by
   multiple `goroutines
   <https://www.golang-book.com/books/intro/10>`__.
   Additionally, you should not use custom implementations of the
   ``Session`` interface in production.
```